saas/mixins.py

```diff
@@ -1219,6 +1220,43 @@ def get_object(self):
             raise Http404('No %s matches the given query.'
                 % queryset.model._meta.object_name)
 
+class BalancesDueMixin(DateRangeContextMixin, ProviderMixin):
+
+    filter_backends = (DateRangeFilter,)
+
+    @property
+    def balances_due(self):
+        if not hasattr(self, "_balances_due"):
+            self._balances_due = get_balances_due(provider=self.provider)
+        return self._balances_due
+
+    def get_queryset(self):
+        organization_model = get_organization_model()
+        if self.provider:
+            queryset = organization_model.objects.filter(
+                subscribes_to__organization=self.provider).distinct()
+        else:
+            queryset = organization_model.objects.all()
+        decorated_queryset = self.decorate_queryset(queryset)
+        return decorated_queryset
+
+    def decorate_queryset(self, queryset):
+        # Decorating the queryset to "attach" the balance values
+        # to the queryset so we can use the OrganizationSerializer
+        organization_ids = []
+        for organization in queryset:
+            customer = self.balances_due.get(organization.slug)
+            if customer:
+                if any(val['balance'] > 0 for val in six.itervalues(customer)):
+                    organization_ids.append(organization.id)
+        decorated_queryset = queryset.filter(id__in=organization_ids).order_by('full_name')
```

Comment:
I would have expected the `balances_due` directly added as an attribute on the Organization (ex: https://github.com/djaodjin/djaodjin-saas/blob/d2ab75d184822b5ee2b2850b388e0386e86a464e/saas/api/metrics.py#L592). It seems a lot of indirection through dictionaries in order to return a `Quesryset` instead of a list.
Maybe the raw SQL in `get_queryset` should be rewritten to include the `val['balance'] > 0` clause and all `Organization` fields such that `get_queryset` returns `Organization.objects.raw(...)`.